komodo-relational/src/main/java/org/komodo/relational/teiid/CachedTeiid.java

```diff
@@ -204,6 +221,36 @@ public CachedTeiid resolve( final UnitOfWork transaction,
     Driver getDriver(final UnitOfWork transaction, String name) throws KException;
 
     /**
+     * Removes a Vdb from the cache, if present
+     * @param transaction
+     *        the transaction (cannot be <code>null</code> or have a state that is not {@link State#NOT_STARTED})
+     * @param name the Vdb name
+     * @throws KException
+     *         if an error occurs
+     */
+    void removeVdb(final UnitOfWork transaction, String name) throws KException;
+
+    /**
+     * Removes a dataSource from the cache, if present
+     * @param transaction
+     *        the transaction (cannot be <code>null</code> or have a state that is not {@link State#NOT_STARTED})
+     * @param name the dataSource name
+     * @throws KException
+     *         if an error occurs
+     */
+    void removeDataSource(final UnitOfWork transaction, String name) throws KException;
+
+    /**
+     * Removes a driver from the cache, if present
+     * @param transaction
+     *        the transaction (cannot be <code>null</code> or have a state that is not {@link State#NOT_STARTED})
+     * @param name the driver name
+     * @throws KException
+     *         if an error occurs
+     */
+    void removeDriver(final UnitOfWork transaction, String name) throws KException;
+
+    /**
      * @param uow
```

Comment:
I really don't like the inclusion of these methods as they invite the possibility of the CachedTeiid object no longer being a cached-version of the teiid state. It makes sense to try and avoid a full refresh for the sake of performance but can that be done 'internally' to the CachedTeiidImpl rather than utilising these methods. My concern is that once they are permanent in the API, other code can come along later and use them, changing the CachedTeiid state and the cache is no longer reflective of the true state of the real teiid instance, which is/should be its only purpose.